internal/app/app.go

```diff
@@ -0,0 +1,71 @@
+package app
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rawen554/shortener/internal/config"
+	"github.com/rawen554/shortener/internal/store"
+	"github.com/rawen554/shortener/internal/utils"
+)
+
+type App struct {
+	config *config.ServerConfig
+	store  *store.Storage
+}
+
+func NewApp(config *config.ServerConfig, storage *store.Storage) *App {
+	return &App{
+		config: config,
+		store:  storage,
+	}
+}
+
+func (a *App) RedirectToOriginal(c *gin.Context) {
+	res := c.Writer
+	id := c.Param("id")
+
+	originalURL := a.store.Get(id)
+
+	if originalURL == nil {
+		res.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	res.Header().Set("Location", string(originalURL))
+	res.WriteHeader(http.StatusTemporaryRedirect)
+}
+
+func (a *App) ShortenURL(c *gin.Context) {
+	req := c.Request
+	res := c.Writer
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		fmt.Printf("body cannot be read: %s\n", err)
```

Comment:
Логирование лучше делать через библиотеку `log`, для первого спринта этого достаточно:

```suggestion
		log.Printf("body cannot be read: %v", err)
```